cmd/fdsn-ws/fdsn_station.go

```diff
@@ -618,10 +626,11 @@ func (r *FDSNStationXML) doFilter(params []fdsnStationV1Search) bool {
 		}
 	}
 
+	nNet := len(r.Network)
 	r.Network = ns
 
-	if len(ns) == 0 {
-		// No result ( no "Network" node )
+	if nNet != 0 && len(ns) == 0 {
+		// this means the query has network parameter but idn't match one, we want to give a no data
 		return false
 	}
```

Comment:
`nNet` is strange.
Can `ns` be `networks` ? or better yet `filteredNetworks`.

Would a `defer r.Network = ns` help, just after for loop? Also for station/channel sections.

Comment missing a `d` in `didn't`

Reply:
There's still `trimLevel` need to operate on the result slice. So can't move the assignment to defer.